spyderlib/restart_app.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright © 2015 The Spyder Development Team
+# Licensed under the terms of the MIT License
+# (see spyderlib/__init__.py for details)
+
+"""
+Restart Spyder
+
+A helper script that allows Spyder to restart from within the application.
+"""
+
+import ast
+import os
+import os.path as osp
+import subprocess
+import sys
+import time
+
+from spyderlib.py3compat import to_text_string

```

Comment:
I just tested this PR and it's not working with `bootstrap` because of this line.

The thing is this assumes Spyder is installed in your virtualenv or globally. But that's almost always not true (at least for me :-), when you're using `bootstrap`.

The only alternative I see is to copy/paste `to_text_string` here and use it below. It's a simple function, so I don't see any problem with that :-)